modules/client/src/main/java/org/apache/ignite/internal/client/compute/ClientCompute.java

```diff
@@ -442,6 +473,26 @@ private static SubmitResult unpackSubmitResult(PayloadInputChannel ch) {
         return new SubmitResult(ch.in().unpackUuid(), ch.notificationFuture());
     }
 
+    /**
+     * Unpacks coordination job id and jobs ids which are executing under this task from channel and gets notification future.
+     * This is needed because we need to unpack message response in the payload
+     * reader because the unpacker will be closed after the response is processed.
+     *
+     * @param ch Payload channel.
+     * @return Result of the task submission.
+     */
+    private static SubmitTaskResult unpackSubmitTaskResult(PayloadInputChannel ch) {
+        var jobId = ch.in().unpackUuid();
+
+        var size = ch.in().unpackInt();
+        List<UUID> jobIds = new ArrayList<>(size);
+        for (int i = 0; i < size; i++) {
+            jobIds.add(ch.in().unpackUuidNullable());
```

Comment:
```suggestion
            jobIds.add(ch.in().unpackUuid());
```

I think the server does not return nulls anymore.